### DISCUSSION TITLE
Utilizing Plot Events of G2 Plot in React Component

### DISCUSSION THREAD

**Discussion by jagijan** [open-data-plan/g2plot-react#354] [Q&A] [answered]
Hello folks,

Could someone provide insight on how to utilize plot events provided in G2 Plot when using g2plot-react library?
I've designed a bar chart & I would like to perform some action when user clicks a particular element (i.e. bar) in the chart.

Thank you in advance!

**Comment by kagawagao** [comment, marked as answer]
you can use `chartRef` to access chart `plot` instance, then follow `g2plot` events [api](https://g2plot.antv.vision/en/docs/api/options/events) to bind event

```ts
  const lineChartRef = useRef<Line>();
  useEffect(() => {
    if (lineChartRef.current) {
      const plot = lineChartRef.current;
      plot.on("line:click", (e) => {
        console.log(e);
      });
    }
  }, []);
```

[demo](https://codesandbox.io/s/g2plot-react-example-xx3gp?fontsize=14&hidenavigation=1&theme=dark&file=/src/line.tsx:4123-4303)
